bam_markdup.c

```diff
@@ -635,10 +632,35 @@ static void make_single_key(key_data_t *key, bam1_t *bam) {
         orientation = O_FF;
     }
 
+    if (param->barcode) {
+        if ((data = bam_aux_get(bam, param->barcode))) {
+            if (!(bar = bam_aux2Z(data))) {
+                fprintf(stderr, "[markdup] error: %s tag wrong type. Aux tag needs to be a string type.\n", param->barcode);
+            }
+
+            barcode = do_hash((unsigned char *)bar, strlen(bar));
+        }
+    } else if (param->bc_rgx) {
+        regmatch_t matches[3];
+        size_t max_matches = 2;
+        char *qname = bam_get_qname(bam);
+        int bc_start, bc_end;
+
+        regexec(param->bc_rgx, qname, max_matches, matches, 0);
+
+        bc_start = matches[1].rm_so;
```

Comment:
`matches[1]` could be use uninitialised here, if the match fails, and then the invalid values may end up passed to `do_hash`.

Reply:
Okay, fair point.  Those are only set to -1 when the string matches but the substring does not.